packages/cornucopia-store-better-sqlite/src/store.ts

```diff
@@ -95,21 +83,34 @@ CREATE TABLE IF NOT EXISTS "${this.table}" (
     });
   }
 
-  public async transaction(
-    selector: ProofSelector<StartProofSelection>,
-    action: Transaction,
-    options?: TransactionOptions,
-  ) {
-    const selection: StartProofSelection = {
-      getAvailableProofs: async () => this.locker.getUnlocked(this.loadAllProofs()),
-      recoverProofs: async () => {
-        const all = this.loadAllProofs();
-        const unlocked = await this.locker.getUnlocked(all);
-        return all.filter((p) => !unlocked.includes(p));
-      },
-    };
-
-    const proofs = await selector(selection);
+  private loadAllProofs() {
+    return this.sqlite.prepare<[], ProofRow>(`SELECT * FROM "${this.table}"`).all().map(convertRowToProof);
+  }
+
+  private async selectProofs(filter: ProofFilter) {
+    let proofs: Proof[];
+    if (filter.amount) {
+      const available = this.sqlite
+        .prepare<[number], ProofRow>(`SELECT * from "${this.table}" WHERE amount <= ?`)
+        .all(filter.amount)
+        .map(convertRowToProof);
+
+      const { send } = selectProofsToSend(available, filter.amount);
+      proofs = send;
+      if (sumProofs(proofs) < filter.amount) throw new Error("Failed to find enough proofs");
+    } else proofs = this.loadAllProofs();
+
+    const unlocked = await this.locker.getUnlocked(proofs);
+    if (filter.locked) proofs = proofs.filter((p) => !unlocked.includes(p));
+    else proofs = unlocked;
+
+    if (filter.expired) proofs = await this.locker.getExpired(proofs);
```

Comment:
Have you thought about the order that these selections occur? I think for the filters you currently have they would mostly be used exclusively. 
ie. Currently if I set amount and expired in my filter, then I might not get all of the expired proofs that exist because some will have already been filtered out. In reverse, the amount selector might get the correct amount, then filter some out